tests/test-cases/Group11-Upgrade/11-01-Upgrade.robot

```diff
@@ -128,5 +119,56 @@ Upgrade VCH with containers
     Should Be Equal As Integers  ${rc}  0
     Should Not Contain  ${output}  Error
 
+Create Docker Containers
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} network create bar
+    Should Be Equal As Integers  ${rc}  0
+    Comment  Launch container on bridge network
+    ${id1}  ${ip1}=  Launch Container  vch-restart-test1  bridge
+    ${id2}  ${ip2}=  Launch Container  vch-restart-test2  bridge
+    Set Environment Variable  ID1  ${id1}
+    Set Environment Variable  ID2  ${id2}
+    Set Environment Variable  IP1  ${ip1}
+    Set Environment Variable  IP2  ${ip2}
+
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} create -it -p 10000:80 -p 10001:80 --name webserver nginx
+    Should Be Equal As Integers  ${rc}  0
+    Should Not Contain  ${output}  Error
+    ${rc}  ${output}=  Run And Return Rc And Output  docker %{VCH-PARAMS} start webserver
+    Should Be Equal As Integers  ${rc}  0
+    Should Not Contain  ${output}  Error
+    Wait Until Keyword Succeeds  20x  5 seconds  Hit Nginx Endpoint  %{VCH-IP}  10000
+    Wait Until Keyword Succeeds  20x  5 seconds  Hit Nginx Endpoint  %{VCH-IP}  10001
+
+*** Test Cases ***
+Upgrade Present in vic-machine
+    ${rc}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux
+    Should Contain  ${output}  upgrade
+
+Upgrade VCH with unreasonably short timeout and automatic rollback after failure
+    Log To Console  \nUpgrading VCH with 1s timeout ...
+    ${rc}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux upgrade --debug 1 --name=%{VCH-NAME} --target=%{TEST_URL} --user=%{TEST_USERNAME} --password=%{TEST_PASSWORD} --force=true --compute-resource=%{TEST_RESOURCE} --timeout 1s
+    Should Contain  ${output}  Upgrading VCH exceeded time limit
+    Should Not Contain  ${output}  Completed successfully
+    ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc snapshot.tree -vm=%{VCH-NAME}/%{VCH-NAME}
+    Run Keyword If  '%{HOST_TYPE}' == 'VC'  Should Not Contain  ${output}  upgrade
+    ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc snapshot.tree -vm=%{VCH-NAME}
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Should Not Contain  ${output}  upgrade
+
+Upgrade VCH
+
+    Create Docker Containers
+    Log To Console  \nUpgrading VCH...
+
+    Upgrade
+    Check Upgraded Version
+
+    Rollback
+    Check Original Version
```

Comment:
fundamentally I am confused why we have a "check upgrade version" and a "check original version" here?  I would think it would just be Upgrade -> Check version -> roll back -> check version -> upgrade -> check version...